integration-tests/ide-client.test.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
+import * as net from 'node:net';
+import { IdeClient } from '../packages/core/src/ide/ide-client.js';
+import { vi } from 'vitest';
+import { getIdeProcessId } from '../packages/core/src/ide/process-utils.js';
+
+vi.mock('../packages/core/src/ide/process-utils', () => ({
+  getIdeProcessId: vi.fn(),
```

Comment:
this doesn't test process-utils as it is mocking it out. Can you make the integration test actually test that logic as it is the most complex logic added. That might require having this integration test intentionally start up a few processes for process-utils to traverse.